plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -829,8 +838,24 @@ def download_rpms(self, metadata_files, rpms_to_download, url):
 
             if self.download_deferred:
                 catalog = PackageCatalog(self.conduit.importer_object_id, url)
+                importer = Importer.objects.get(id=self.conduit.importer_object_id)
+
                 for unit in units_to_download:
                     unit.downloaded = False
+                    # If feed-url has changed since last-sync, we need to:
+                    #   - delete existing matching-unit if there is one, and
+                    #   - delete existing LazyCatalogEntry if there is one,
+                    # in order to safely rebuild LCE with possibly-new download-urls
+                    # [NOTE: "if there is one" is almost certainly paranoia at this point in the
+                    # code-flow. Paranoia is not always invalid, however...]
+                    if importer_constants.KEY_FEED_UPDATED in importer.config and \
+                            importer.config[importer_constants.KEY_FEED_UPDATED]:
+                        units = unit.__class__.objects.filter(**unit.unit_key)
+                        repo_controller.disassociate_units(self.repo, units)
+                        for old_unit in units:
+                            old_unit.delete()
+                        catalog.delete(unit)
```

Comment:
even if calling delete on the lazy entry, i can still see it after sync is complete.

Reply:
Hm - do have an exact reproducer? I actively tried to make that happen and failed, I must be missing something.